### PR TITLE
[PnP] Fixing bug that broke integ tests

### PR DESCRIPTION
**Pull request by dylanbronson**
The last pnp PR broke our integ tests because of two subtle bugs.

First is that the metadataStore needs to be mocked in two more places.
2nd is that I had a more stringent check in MetadataStore that broke some of the integ tests, since the integ tests use empty string as edgeProductInfo.

As I don't want to cause any breaking changes, I changed the metadataStore check to be the same as it was before (CheckNotNull instead of CheckNonWhitespace)

### CODE REVIEW

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionMetadata.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ public ConnectionMetadata(string productInfo, string modelId, string edgeProduct
 
         public ConnectionMetadata(string productInfo, Option<string> modelId, string edgeProductInfo)
         {
-            Preconditions.CheckNonWhiteSpace(edgeProductInfo, nameof(edgeProductInfo));
```
Comment (and-rewsmith) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionMetadata.cs:21]: You said you didn't want to cause breaking changes, right? Why can't we change the tests to send a edgeProductInfo?

File: edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/EdgeHubConnectionTest.cs [hunk 2]
```diff
@@ -62,7 +62,8 @@ public async Task TestEdgeHubConnection()
             string sasKey = ConnectionStringHelper.GetSharedAccessKey(deviceConnStr);
             var signatureProvider = new SharedAccessKeySignatureProvider(sasKey);
             var credentialsCache = Mock.Of<ICredentialsCache>();
-            var metadataStore = Mock.Of<IMetadataStore>();
+            var metadataStore = new Mock<IMetadataStore>();
+            metadataStore.Setup(m => m.GetMetadata(It.IsAny<string>())).ReturnsAsync(new ConnectionMetadata("dummyValue"));
```
Comment (and-rewsmith) [inline edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/EdgeHubConnectionTest.cs:66]: Why is this change needed?

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionMetadata.cs (hunk 1 above)
Comment (and-rewsmith) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionMetadata.cs:21]: Are you concerned about breaking changes from 1.0.10 RC to 1.0.10?

Comment (bronzeson) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionMetadata.cs:21]: The breaking change I'm concerned about is that the old ProductInfo allows Empty string/whitespace as edge product info, but the new MetadataStore does not. This change relaxes it back to how the old ProductInfoStore had it.
So it's breaking change between 1.0.10 RC-1 -> 1.0.10 RC-2 and above

File: edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/EdgeHubConnectionTest.cs (hunk 2 above)
Comment (bronzeson) [reply edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/EdgeHubConnectionTest.cs:66]: The metadata store object needs to be mocked because there is a GetMetadata in the new flow.
This GetMetadata is safe and will always return a ConnectionMetadata object, but if we don't mock it, the GetMetadata call fails and the test breaks

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionMetadata.cs (hunk 1 above)
Comment (dylanbronson) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionMetadata.cs:21]: The breaking change I'm concerned about is that the old ProductInfo allows Empty string/whitespace as edge product info, but the new MetadataStore does not. This change relaxes it back to how the old ProductInfoStore had it.
So it's breaking change between 1.0.10 RC-1 -> 1.0.10 RC-2 and above

File: edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/EdgeHubConnectionTest.cs (hunk 2 above)
Comment (dylanbronson) [reply edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/EdgeHubConnectionTest.cs:66]: The metadata store object needs to be mocked because there is a MetadataStore.GetMetadata in the new flow.
This GetMetadata is safe and will always return a ConnectionMetadata object, but if we don't mock it, the GetMetadata call fails and the test breaks
